### PR TITLE
Rwc2019 say

### PR DESCRIPTION
**Pull request by MatthijsBurgh**

### CODE REVIEW

File: robot_smach_states/src/robot_smach_states/human_interaction/human_interaction.py [hunk 1]
```diff
@@ -81,6 +88,93 @@ def execute(self, userdata=None):
...
+
+    def _check_place_holders(self, sentence):
+        if isinstance(sentence, list):
+            for sen in sentence:
+                self._check_place_holders(sen)
+            return
+
+        place_holders = [x[1] for x in Formatter().parse(sentence) if x[1] is not None]
+        if not all(x in place_holders for x in self.ph_designators.keys()):
+            raise RuntimeError("Not all named place holders are provided")
```
[the remark this answers is not in the document]
Comment (MatthijsBurgh) [reply robot_smach_states/src/robot_smach_states/human_interaction/human_interaction.py]: Fixed

File: robot_smach_states/src/robot_smach_states/human_interaction/human_interaction.py [hunk 2]
```diff
@@ -81,6 +88,93 @@ def execute(self, userdata=None):
...
+        ds.check_type(sentence, str, list)
+        assert(isinstance(language, str) or isinstance(language, type(None)))
+        assert(isinstance(personality, str) or isinstance(personality, type(None)))
+        assert(isinstance(voice, str) or isinstance(voice, type(None)))
+        assert(isinstance(mood, str) or isinstance(mood, type(None)))
+        assert(isinstance(block, bool))
+
+        assert(all(isinstance(v, ds.Designator) for v in kwargs.values()))
+
+        self.ph_designators = kwargs
```
[the remark this answers is not in the document]
Comment (MatthijsBurgh) [reply robot_smach_states/src/robot_smach_states/human_interaction/human_interaction.py]: done

File: robot_skills/src/robot_skills/mockbot.py [hunk 3]
```diff
@@ -281,9 +281,7 @@ class Mockbot(robot.Robot):
     >>> Mockbot()
     """
     def __init__(self, *args, **kwargs):
-        robot_name = "mockbot"
-
-        super(Mockbot, self).__init__(robot_name="mockbot", wait_services=False)
+        super(Mockbot, self).__init__(robot_name="robot_name", wait_services=False)
```
Comment (LoyVanBeek) [inline robot_skills/src/robot_skills/mockbot.py]: Is that robot_name correct?

Comment (MatthijsBurgh) [reply robot_skills/src/robot_skills/mockbot.py]: fixed

**Pull request merged**